pkg/auth/service.go

```diff
@@ -959,3 +964,626 @@ func (s *DBAuthService) Authorize(ctx context.Context, req *AuthorizationRequest
 	// we're allowed!
 	return &AuthorizationResponse{Allowed: true}, nil
 }
+
+type APIAuthService struct {
+	apiClient   *ClientWithResponses
+	secretStore crypt.SecretStore
+	cache       Cache
+}
+
+func (a *APIAuthService) SecretStore() crypt.SecretStore {
+	return a.secretStore
+}
+
+func (a *APIAuthService) CreateUser(ctx context.Context, user *model.User) (int, error) {
+	resp, err := a.apiClient.CreateUserWithResponse(ctx, CreateUserJSONRequestBody{
+		Username: user.Username,
+	})
+	if err != nil {
+		return InvalidUserID, err
+	}
+	if err := expectedStatusCode(resp, 201); err != nil {
```

Comment:
As I see it we can have one function that handles all the remote api return values (errors) and map them to the local error. Alternative will be to have a switch case per status code to handle the local error mapping.